crowdin.js

```diff
@@ -0,0 +1,156 @@
+const AdmZip = require('adm-zip');
+const program = require('commander');
+const request = require('request');
+const fs = require('fs');
+
+/* eslint-disable no-console */
+
+function help() {
+	console.log('To use crowdin script to get the status: ');
+	console.log('>node crowding.js --status');
+	console.log('To use crowdin script to upload current catalog: ');
+	console.log('>node crowding.js --upload ./i18n/components/tui-components.json');
+}
+
+program
+	.version('0.0.1')
+	.option('-k, --key [value]', 'REQUIRED PROJECT KEY')
+	.option('-v, --verbose', 'display debug info')
+	.option('-d, --download', 'download all the translations')
+	.option('-s, --status', 'get the status of translations')
+	.option('-u, --upload [value]', 'upload translation')
+	;
+
+program.on('--help', help);
+
+program.parse(process.argv);
+
+function debug(...args) {
+	if (program.verbose) {
+		console.log(...args);
+	}
+}
+
+// TODO: prompt for the PROJECT_KEY
+
+const PROJECT_KEY = program.key;
+const PROJECT_ID = 'talendui';
+const URL = `https://api.crowdin.com/api/project/${PROJECT_ID}`;
+
+const FILES = [
+	{
+		name: 'tui-components.json',
+		path: `${__dirname}/i18n/components/en/tui-components.json`,
+		locales: `${__dirname}/components/locales`,
+	},
+];
+
+function onError(error) {
+	if (error) {
+		throw error;
+	}
+}
+
+function handleStatusRequest(error, response, body) {
+	onError(error);
+	console.log('---------- status ---------');
+	console.log(body);
+}
+
+function status() {
+	const url = `${URL}/status?json&key=${PROJECT_KEY}`;
+	debug(`status send GET , ${url}`);
+	request(url, handleStatusRequest);
+}
+
+function getDestination(fileEntry) {
+	debug(fileEntry.toString());
+	const path = fileEntry.entryName;
+	if (fileEntry.name === 'tui-components.json') {
+		return `packages/components/locales/${path}`; // "ja/tui-components.json"
+	} else if (fileEntry.name === 'tui-forms.json') {
+		return `packages/forms/locales/${path}`; // "ja/tui-forms.json"
+	} else if (fileEntry.isDirectory) {
+		debug(`directory found supposed to be a language folder: ${path}`);
+		if (!fs.existsSync(`packages/components/locales/${path}`)) {
```

Comment:
`getDestination()` should not create anything, but just return a destination.
Perhaps you can move that into `extractAndSave()`. At the beginning, test if it's a folder, if so create the folders, else write.